torch/nested/_internal/ops.py

```diff
@@ -78,6 +78,37 @@ def raggedness_matches(nt, size):
     return list(nt._size[:end]) == list(size[:end])
 
 
+def squeeze_leading_ones_get_extra(t, nt):
+    # Note: [ Squeezing leading ones ]
+    #
+    # Squeeze leading ones from t, and return the number of extra ones
+    # that need to be unsqueezed after relying on dense broadcasting
+    # Today, we would error if extra != 0 because NT does not yet support
+    # unsqueezing on the 0th dimension.
+    #
+    # We want:
+    #   (B, j0, ?, ?) + (1, 1, ?, ?) -> (B, j0, ?, ?)
+    #   (B, j0, ?, ?) + (1, 1, 1, ?, ?) -> (1, B, j0, ?, ?)  (not yet supported)
+    #
+    # 1) Squeeze extra ones and grab values from NT
+    #   (1, 1, ?, ?) -> (?, ?)   and   (sum(*), ?, ?) -> (B, j0, ?, ?)
+    # 2) Do dense broadcasting:
+    #   (sum(*), ?, ?) + (?, ?) -> (sum(*), ?, ?)
+    # 3) Construct nested tensor
+    #   (sum(*), ?, ?) -> (B, j0, ?, ?)
+    #
+    # If unsqueezing on the 0th dim becomes supported, we would unsqueeze
+    # at step (4).
+    t_shape = t.shape
+    extra = 0
+    for s in t_shape:
+        if s == 1:
+            if t.dim() > nt.dim():
+                extra += 1
+            t = t.squeeze(0)
+    return t, extra
```

Comment:
does this all work within torch.compile without unwanted guards?

Reply:
Hm let me check

Reply:
I do think we do want the extra guards here.

Reply:
hm okay, which guards show up?

Reply:
> I do think we do want the extra guards here.

Whoops, this is wrong actually. What is happening is that there ARE always guards testing whether inputs given are zero/one or not (due to zero-one specialization) which is indeed what we want, but also, as a result of that there are also no "extra" guards from this `s == 1` test here.